src/routes/bounce_details_page.jsx

```diff
@@ -56,10 +82,22 @@ export default class BounceDetailsPage extends React.Component {
     if (status === 200) {
       this.setState({
         currentRule: data,
+        socketConnection: ws,
+      });
+    } else {
+      this.setState({
+        socketConnection: ws,
       });
     }
   }
 
+  componentWillUnmount() {
+    const { currentRule, editText, socketConnection } = this.state;
```

Comment:
In the componentWillUnmount we should also remove any event listeners to avoid any memory leaks. if a user keeps on checking out the page and adding event listeners we need to also remove them on unmount to be sure we don't have a lot of duplicate event listeners

Reply:
Same to double check for the ws stuff is getting cleaned up